Beans without package causes compilation error

Test classes without package in code definition marked with @MicronautTest annotation causes compilation error

Example
```
import io.micronaut.test.annotation.MicronautTest;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.Test;

@MicronautTest
public class Example
{
    @Test
    void doTest()
    {
        Assertions.assertNotNull(null);
    }
}
```
gradle test output
```
> Task :compileTestJava FAILED
Note: Creating bean classes for 2 type elements
error: Unexpected error: Illegal name .$ExampleDefinition
1 error
FAILURE: Build failed with an exception.
```

workarround: put class in any package

--- a/inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy
+++ b/inject-groovy/src/main/groovy/io/micronaut/ast/groovy/InjectTransform.groovy
@@ -1535,6 +1535,10 @@ class InjectTransform implements ASTTransformation, CompilationUnitAware {
 
         private void defineBeanDefinition(ClassNode classNode) {
             if (!beanDefinitionWriters.containsKey(classNode)) {
+                if (classNode.packageName == null) {
+                    addError("Micronaut beans cannot be in the default package", classNode)
+                    return
+                }
                 ClassNode providerGenericType = AstGenericUtils.resolveInterfaceGenericType(classNode, Provider)
                 boolean isProvider = providerGenericType != null
                 AnnotationMetadata annotationMetadata = AstAnnotationUtils.getAnnotationMetadata(sourceUnit, compilationUnit, classNode)

--- a/inject-groovy/src/test/groovy/io/micronaut/inject/context/NoPackageSpec.groovy
+++ b/inject-groovy/src/test/groovy/io/micronaut/inject/context/NoPackageSpec.groovy
@@ -1,0 +1,22 @@
+package io.micronaut.inject.context
+
+import io.micronaut.AbstractBeanDefinitionSpec
+
+class NoPackageSpec extends AbstractBeanDefinitionSpec {
+
+    void "test creating a bean in the default package"() {
+        when:
+        buildBeanDefinition('Bean', '''
+
+import io.micronaut.context.annotation.*
+
+@javax.inject.Singleton
+class Bean {
+
+}
+''')
+        then:
+        def ex = thrown(RuntimeException)
+        ex.message.contains("Micronaut beans cannot be in the default package")
+    }
+}

--- a/inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java
+++ b/inject-java/src/main/java/io/micronaut/annotation/processing/BeanDefinitionInjectProcessor.java
@@ -464,6 +464,11 @@ public class BeanDefinitionInjectProcessor extends AbstractInjectAnnotationProce
 
                         if (isBean) {
                             // we know this class has supported annotations so we need a beandef writer for it
+                            PackageElement packageElement = elementUtils.getPackageOf(classElement);
+                            if (packageElement.isUnnamed()) {
+                                error(classElement, "Micronaut beans cannot be in the default package");
+                                return null;
+                            }
                             BeanDefinitionVisitor beanDefinitionWriter = getOrCreateBeanDefinitionWriter(classElement, qualifiedName);
 
                             if (isAopProxyType) {

--- a/inject-java/src/test/groovy/io/micronaut/inject/context/NoPackageSpec.groovy
+++ b/inject-java/src/test/groovy/io/micronaut/inject/context/NoPackageSpec.groovy
@@ -1,0 +1,22 @@
+package io.micronaut.inject.context
+
+import io.micronaut.inject.AbstractTypeElementSpec
+
+class NoPackageSpec extends AbstractTypeElementSpec {
+
+    void "test creating a bean in the default package"() {
+        when:
+        buildBeanDefinition('Bean', '''
+
+import io.micronaut.context.annotation.*;
+
+@javax.inject.Singleton
+class Bean {
+
+}
+''')
+        then:
+        def ex = thrown(RuntimeException)
+        ex.message.contains("Micronaut beans cannot be in the default package")
+    }
+}

